api/src/main/resources/workbench.yaml

```diff
@@ -457,6 +457,78 @@ paths:
           schema:
             $ref: '#/definitions/ErrorReport'
 
+  /api/v1/workspaces/{workspaceNamespace}/{workspaceId}/notebook-list/:
+    get:
+      summary: Get details of Python files from google Bucket directory notebook
+      description: Returns list of name and path of python files from google bucket, directory notebook.
+      operationId: getNoteBookList
+      tags:
+        - workspaces
+      parameters:
+        - in: path
+          name: workspaceNamespace
+          description: workspaceNamespace
+          required: true
+          type: string
+        - in: path
+          name: workspaceId
+          description: workspaceId
+          required: true
+          type: string
+      responses:
+        200:
+          description: List of files
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/FileDetail'
+        404:
+          description: Workspace not found
+          schema:
+            $ref: '#/definitions/ErrorReport'
+        500:
+          description: Internal Error
+          schema:
+            $ref: '#/definitions/ErrorReport'
+
+  /api/v1/workspaces/{workspaceNamespace}/{workspaceId}/localize-notebook/:
+    post:
+      summary: Localize files from workspace to notebook cluster
+      description: localize files
+      operationId: localizeNotebook
+      tags:
+        - cluster
+      parameters:
+      - in: path
+        name: workspaceNamespace
+        description: workspaceNamespace
+        required: true
+        type: string
+      - in: path
+        name: workspaceId
+        description: workspaceId
+        required: true
+        type: string
+      - in: body
+        name: fileList
+        description: List of file to be transfered
+        required: true
+        schema:
+          type: array
+          items:
+            $ref: '#/definitions/FileDetail'
+      responses:
+        200:
+         description: List of files
+        404:
+         description: Cluster not found
```

Comment:
Or workspace?

Reply:
Not addressed? Disagree?

Reply:
Cluster or workspace not found exception has been added. 
However in case workspace is not valid, the list of notebooks will not be return on the workspace screen hence localize will not happen.  
This has been added if localize api is called by some other method in future